OSDOCS-14603: assembly and module creation for ESO docs (#93201)
openshift/openshift-docs: modules/external-secrets-operator-install-console.adoc

// Module included in the following assemblies:
//
// * security/external_secrets_operator/external-secrets-operator-install.adoc

:_mod-docs-content-type: PROCEDURE
[id="external-secrets-operator-install-console_{context}"]
= Installing the {external-secrets-operator} by using the web console

You can use the web console to install the {external-secrets-operator}.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* You have access to the {product-title} web console.

.Procedure

. Log in to the {product-title} web console.
. Navigate to *Operators* -> *OperatorHub*.

> . Enter *{external-secrets-operator}* into the filter box.

Author: done

. Select the *{external-secrets-operator}*

> . Select the {external-secrets-operator} version from *Version* drop-down list, and click *Install*.

Author: done

//+
//[NOTE]
//====
//place holder for TechPreview release details or See supported {external-secrets-operator} versions in the following "Additional resources" section.
//====

> . On the *Install Operator* page:

Author: done

> .. Update the *Update channel*, if necessary. The channel defaults to *name to be added*, which installs the latest stable release of the {external-secrets-operator}.

Author: done

.. Choose the *Installed Namespace* for the Operator. The default Operator namespace is `external-secrets-operator`.
+
If the `external-secrets-operator` namespace does not exist, it is created for you.
+
.. Select an *Update approval* strategy.
+

> * The *Automatic* strategy allows Operator Lifecycle Manager (OLM) to automatically update the Operator when a new version is available.

Author: done

+
* The *Manual* strategy requires a user with appropriate credentials to approve the Operator update.

> .. Click *Install*.

Author: done

.Verification

. Navigate to *Operators* -> *Installed Operators*.

> . Verify that *{external-secrets-operator}* is listed with a *Status* of *Succeeded* in the `external-secrets-operator` namespace.

Author: done

> . Verify that `external-secrets` pods are up and running by entering the following command:

Author: done

+
[source,terminal]
----

> $ oc get pods -n external-secrets

Reviewer: ```suggestion
$ oc get pods -n external-secrets-operator
```

Author: as per discussion with @bharath-b-rh and @KeenonLee, retaining  `external-secrets` as is.

// Module included in the following assemblies:
//
// * security/external_secrets_operator/external-secrets-operator-install.adoc

:_mod-docs-content-type: PROCEDURE
[id="external-secrets-operator-install-console_{context}"]
= Installing the {external-secrets-operator} by using the web console

You can use the web console to install the {external-secrets-operator}.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* You have access to the {product-title} web console.

.Procedure

. Log in to the {product-title} web console.
. Navigate to *Operators* -> *OperatorHub*.
. Enter *{external-secrets-operator}* into the filter box.
. Select the *{external-secrets-operator}*
. Select the {external-secrets-operator} version from *Version* drop-down list, and click *Install*.
//+
//[NOTE]
//====

> //See supported {cert-manager-operator} versions in the following "Additional resources" section.

Reviewer 2: ```suggestion
//See supported {external-secrets-operator} versions in the following "Additional resources" section.
```

Author: done

//====
. On the *Install Operator* page:

> .. Update the *Update channel*, if necessary. The channel defaults to *stable-v1*, which installs the latest stable release of the {external-secrets-operator}.

Reviewer 2: `stable-v1` channel will have to changed. Since this will be TechPreview, we will use a different channel name. I will share it by end of this week

Author: Since we do not have the channel name currently, I have changed it to *name to be added*. Can we proceed with this now? I can update this in the API PR that I will create later (once I have the API info)

Author: As discussed in slack, going ahead with this suggestion.

.. Choose the *Installed Namespace* for the Operator. The default Operator namespace is `external-secrets-operator`.
+

> If the `cert-manager-operator` namespace does not exist, it is created for you.

Reviewer 2: ```suggestion
If the `external-secrets-operator` namespace does not exist, it is created for you.
```

Author: done

+

> //[NOTE]
> //====
> //During the installation, the {product-title}  web console allows you to select between `AllNamespaces` and `SingleNamespace` installation modes. For installations with {cert-manager-operator} version 1.15.0 or later, it is recommended to choose the `AllNamespaces` installation mode. `SingleNamespace` and `OwnNamespace` support will remain for earlier versions but will be deprecated in future versions.
> //====

Reviewer 2: ```suggestion

```

.. Select an *Update approval* strategy.
+
* The *Automatic* strategy allows Operator Lifecycle Manager (OLM) to automatically update the Operator when a new version is available.
+
* The *Manual* strategy requires a user with appropriate credentials to approve the Operator update.
.. Click *Install*.

.Verification

. Navigate to *Operators* -> *Installed Operators*.

> . Verify that *{external-secrets-operator}* is listed with a *Status* of *Succeeded* in the `cert-manager-operator` namespace.

Reviewer 2: ```suggestion
. Verify that *{external-secrets-operator}* is listed with a *Status* of *Succeeded* in the `external-secrets-operator` namespace.
```

Author: done

// Module included in the following assemblies:
//
// * security/external_secrets_operator/external-secrets-operator-install.adoc

:_mod-docs-content-type: PROCEDURE
[id="external-secrets-operator-install-console_{context}"]
= Installing the {external-secrets-operator} by using the web console

You can use the web console to install the {external-secrets-operator}.

.Prerequisites

* You have access to the cluster with `cluster-admin` privileges.
* You have access to the {product-title} web console.

.Procedure

. Log in to the {product-title} web console.

. Navigate to *Operators* -> *OperatorHub*.

. Enter *{external-secrets-operator-short}* in the search box.

. Select the *{external-secrets-operator}* from the generated list and click *Install*.
//+
//[NOTE]
//====
//place holder for TechPreview release details or See supported {external-secrets-operator-short} versions in the following "Additional resources" section.
//====
. On the *Install Operator* page:

> .. Update the *Update channel*, if necessary. The channel defaults to *name to be added*, which installs the latest stable release of the {external-secrets-operator-short}.

Author: More content is to be added. It will be added in the next PR when the content is ready. I have mentioned the same in Additional information section in this PR.
getting this PR merged with the content that is ready.

.. Choose the *Installed Namespace* for the Operator.
+
* To use the default Operator namespace, select the *Operator recommended Namespace* option.
+
* To use the namespace that you created, select the *Select a Namespace* option, and then select the namespace from the drop-down list.
+
* If the default `external-secrets-operator` namespace does not exist, it is created for you by the {olm-first}.
+
.. Select an *Update approval* strategy.
+
* The *Automatic* strategy enables {olm} to automatically update the Operator when a new version is available.
+
* The *Manual* strategy requires a user with appropriate credentials to approve the Operator update.

.. Click *Install*.

.Verification

Verify the installation by completing the following steps:

. Navigate to *Operators* -> *Installed Operators*.

. Verify that *{external-secrets-operator-short}* is listed with a *Status* of *Succeeded* in the `external-secrets-operator` namespace.

. Verify that `external-secrets` pods are up and running by entering the following command:
+
[source,terminal]
----
$ oc get pods -n external-secrets
----

+
.Example output
[source,terminal]
----

> //to be added

Author: More content is to be added. It will be added in the next PR when the content is ready. I have mentioned the same in Additional information section in this PR.
getting this PR merged with the content that is ready.
